src/lib/base58_check/base58_check.mli

```diff
@@ -18,6 +18,9 @@ end) : sig
    * exceptions and a B58.Invalid_base58_character *)
   val decode_exn : string -> string
 
+  (** Same as decode_exn but with [target] specified in the error message *)
```

Comment:
Instead of having an extra argument here, maybe it would be better if `Base58_check.Make` had a descriptive string in its struct argument that could be used. `Base58_check.Version_bytes` could pair the bytes with such strings.